helix-term/src/ui/editor.rs

```diff
@@ -413,6 +419,92 @@ impl EditorView {
         spans
     }
 
+    pub fn render_sticky_context(
+        editor: &Editor,
+        doc: &Document,
+        view: &View,
+        surface: &mut Surface,
+        theme: &Theme,
+    ) -> Option<Vec<usize>> {
+        if let Some(syntax) = doc.syntax() {
+            let tree = syntax.tree();
+            let text = doc.text().slice(..);
+            let cursor_byte = doc.selection(view.id).primary().cursor(text);
+            let context_nodes = doc
+                .language_config()
+                .and_then(|lc| lc.sticky_context_nodes.as_ref());
+
+            let mut parent = tree
+                .root_node()
+                .descendant_for_byte_range(cursor_byte, cursor_byte)
+                .and_then(|n| n.parent());
+
+            // context is list of numbers of lines that should be rendered in the LSP context
+            let mut context: Vec<usize> = Vec::new();
+
+            while let Some(node) = parent {
+                let line = text.byte_to_line(node.start_byte());
+
+                // if parent of previous node is still on the same line, use the parent node
+                if let Some(&prev_line) = context.last() {
+                    if prev_line == line {
+                        context.pop();
+                    }
+                }
+
+                if context_nodes.map_or(true, |nodes| nodes.iter().any(|n| n == node.kind())) {
+                    context.push(line);
+                }
+
+                parent = node.parent();
+            }
+
+            // we render from top most (last in the list)
+            context.reverse();
+
+            // TODO: this probably needs it's own style, although it seems to work well even with cursorline
+            let context_style = theme.get("ui.cursorline.primary");
+            let mut context_area = view.inner_area();
+            context_area.height = 1;
+
+            let mut line_numbers = Vec::new();
+            for line_num in context {
+                if line_num >= view.offset.row {
```

Comment:
```suggestion
                if line_num >= view.offset.row || text.byte_to_line(cursor_byte) == line_num {
```

this is required because the sticky context will overshadow the cursor if you're on the same line with the cursor.

Reply:
Have you tried with the latest changes? I ammended it to the latest comming but for me it works just with `line_num >= view.offset.row`, previously it was `line_num > view.offset.row`.

Reply:
> Have you tried with the latest changes? I ammended it to the latest commit but for me it works just with `line_num >= view.offset.row`, previously it was `line_num > view.offset.row`.

actually i haven't. time to try it out when i have time :)

Reply:
Update: yup seems good